test/testdata/rewriter/prop_default.rb

```diff
@@ -1,5 +1,5 @@
 # typed: true
 
 class A < T::Struct
-  prop :foo, T.nilable(Integer), default: '' # error: Argument does not have asserted type `T.nilable(Integer)`
+  prop :foo, T.nilable(Integer), default: '' # error-with-dupes: Argument does not have asserted type `T.nilable(Integer)`
```

Comment:
Do you know where the dup is coming from here?

Reply:
Yes, I think we're raising the same error in multiple places now.

Reply:
I believe it's the generated `initialize` method from the `props` dsl pass: it generates both an argument with that default and a use of `T.let`, both of which will raise the error.